incubator/virtualcluster/cmd/manager/main.go

```diff
@@ -46,14 +46,19 @@ func main() {
 		leaderElectionCmName    string
 		maxConcurrentReconciles int
 		versionOpt              bool
+		disableStacktrace       bool
 	)
 	flag.StringVar(&metricsAddr, "metrics-addr", ":0", "The address the metric endpoint binds to.")
-	flag.StringVar(&masterProvisioner, "master-prov", "native", "The underlying platform that will provision master for virtualcluster.")
+	flag.StringVar(&masterProvisioner, "master-prov", "native",
+		"The underlying platform that will provision master for virtualcluster.")
 	flag.BoolVar(&leaderElection, "leader-election", true, "If enable leaderelection for vc-manager")
-	flag.StringVar(&leaderElectionCmName, "le-cm-name", "vc-manager-leaderelection-lock", "The name of the configmap that will be used as the resourcelook for leaderelection")
-	flag.IntVar(&maxConcurrentReconciles, "num-reconciles", 10, "The max number reconcilers of virtualcluster controller")
+	flag.StringVar(&leaderElectionCmName, "le-cm-name", "vc-manager-leaderelection-lock",
+		"The name of the configmap that will be used as the resourcelook for leaderelection")
+	flag.IntVar(&maxConcurrentReconciles, "num-reconciles", 10,
+		"The max number reconcilers of virtualcluster controller")
 	flag.StringVar(&logFile, "log-file", "", "The path of the logfile, if not set, only log to the stderr")
 	flag.BoolVar(&versionOpt, "version", false, "Print the version information")
+	flag.BoolVar(&disableStacktrace, "disable-stacktrace", false, "If set, the automatic stacktrace is disabled")
```

Comment:
Does being able to change the log level change whether the stacktraces happen with `zap`? I'm curious since this is expanding the flags we'd need to continue to support if we could make something more extensible like `--log-level=xyz` would give the same benefit but allow more control…

Reply:
By default, the zap captures the stackstrace for ErrorLevel and above loge-levels in production. This is non-configurable, the only option is disabling the stackstrace for all log-level.

Reply:
Ah, okay that makes more sense. Happy to leave it this way then.